checkstyle-tester/README.md

```diff
@@ -1,138 +1,59 @@
-
-In order to generate a compact diff report before and after your changes you can use diff.groovy script which performs all required work automatically. Note, diff.groovy ignores excludes specified in the list of projects file.
-Please execute the following command in your command line to run diff.groovy:
+If you want to specify different Checkstyle configs for base branch and patch branch use the following command:
 
 ```
-groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --baseBranch master --patchBranch i111-my-fix --config my_check.xml --listOfProjects projects-to-test-on.properties
+groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --baseBranch master --patchBranch i111-my-fix --baseConfig base_config.xml --patchConfig patch_config.xml --listOfProjects projects-to-test-on.properties
 ```
 
 or with short command line arguments names:
 
 ```
-groovy diff.groovy -r /home/johndoe/projects/checkstyle -b master -p i111-my-fix -c my_check.xml -l projects-to-test-on.properties
+groovy diff.groovy -r /home/johndoe/projects/checkstyle -b master -p i111-my-fix -bc base_config.xml -pc patch_config.xml -l projects-to-test-on.properties
 ```
 
-If you want to specify different Checkstyle configs for base branch and patch branch use the following command:
+To generate the report only for the patch branch which contains your changes, use the following command:
 
 ```
-groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --baseBranch master --patchBranch i111-my-fix --baseConfig base_config.xml --patchConfig patch_config.xml --listOfProjects projects-to-test-on.properties
+groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --patchBranch i111-my-fix --patchConfig patch_config.xml --listOfProjects projects-to-test-on.properties --patchOnly
 ```
 
 or with short command line arguments names:
 
 ```
-groovy diff.groovy -r /home/johndoe/projects/checkstyle -b master -p i111-my-fix -bc base_config.xml -pc patch_config.xml -l projects-to-test-on.properties
+groovy diff.groovy -r /home/johndoe/projects/checkstyle -p i111-my-fix -pc patch_config.xml -l projects-to-test-on.properties -po
 ```
 
 The script receives the following set of command line arguments:
 
 **localGitRepo** (r) - path to the local Checkstyle repository (required);
 
-**baseBranch** (b) - name of the base branch in local Checkstyle repository (optional, default is master branch);
+**baseBranch** (b) - name of the base branch in local Checkstyle repository (optional, if asent, then the tool will use only patchBranch in case patchOnly is set, otherwise baseBrach will be set to 'master');
+
+**patchOnly** (po) - whether to generate report only for the patch branch. If the option is set, then baseBranch, baseConfig and config should not be specified as the tool will finish the execution with the error. You must specify patchBranch and patchConfig if you enable this option. (optional, not specified by default)
```

Comment:
romani wants the option to be named `mode` with the values `diff` (default) or `single`.
https://github.com/checkstyle/contribution/issues/213#issuecomment-297531342

I think we should explicitly tell users why they would use this option.
This is for any option that can't work on master/base branch, which are new properties, new tokens, or new modules.